apps/meteor/client/views/omnichannel/departments/DepartmentsRoute.js

```diff
@@ -1,19 +1,19 @@
 import { Table } from '@rocket.chat/fuselage';
 import { useDebouncedValue, useMutableCallback } from '@rocket.chat/fuselage-hooks';
-import { useRouteParameter, useRoute, usePermission, useTranslation } from '@rocket.chat/ui-contexts';
-import React, { useMemo, useCallback, useState } from 'react';
+import { useRouteParameter, useRoute, usePermission, useTranslation, useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import React, { useMemo, useCallback, useState, useRef } from 'react';
 
 import GenericTable from '../../../components/GenericTable';
-import { useEndpointData } from '../../../hooks/useEndpointData';
 import NotAuthorizedPage from '../../notAuthorized/NotAuthorizedPage';
 import DepartmentsPage from './DepartmentsPage';
-import EditDepartment from './EditDepartment';
 import EditDepartmentWithData from './EditDepartmentWithData';
+import NewDepartment from './NewDepartment';
 import RemoveDepartmentButton from './RemoveDepartmentButton';
 
 const sortDir = (sortDir) => (sortDir === 'asc' ? 1 : -1);
 
-const useQuery = ({ text, itemsPerPage, current }, [column, direction], onlyMyDepartments) =>
+const useDepartmentsQuery = ({ text, itemsPerPage, current }, [column, direction], onlyMyDepartments) =>
```

Comment:
`nitpick` the name conflicts with our new standard of suffixing hooks enclosing React Query's `useQuery`.

```suggestion
const useDepartmentsParams = ({ text, itemsPerPage, current }, [column, direction], onlyMyDepartments) =>
```